openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -806,6 +924,138 @@ def match_volume_by_prefix(self, prefix):
         match_heketi_and_gluster_volumes_by_prefix(
             h_vol_names, "{}_".format(prefix))
 
+    def cleanup_heketi_volumes(
+            self, initial_vols, raise_on_error=False):
+        """Delete heketi volumes and blockvolumes.
+
+        Args:
+            initial_vols (list): Volumes present at initial
+        """
+        h_node, h_url = self.heketi_client_node, self.heketi_server_url
+
+        final_vols = heketi_volume_list(
+            h_node, h_url, json=True).get("volumes")
+        diff_volumes = list(set(final_vols) - set(initial_vols))
+        for volume in diff_volumes:
+            h_vol_info = heketi_volume_info(h_node, h_url, volume, json=True)
+            if h_vol_info.get("block", {}):
+                for block_vol in (
+                        h_vol_info.get(
+                            "blockinfo", {}).get("blockvolume", {})):
+                    heketi_blockvolume_delete(h_node, h_url, block_vol)
+            heketi_volume_delete(
+                h_node, h_url, volume, raise_on_error=raise_on_error)
+
+    def create_heketi_blockvolumes_in_batch(
+            self, vol_count, size=1, batch_amount=8, timeout=300, interval=5,
+            skip_cleanup=False, **kwargs):
+        """Creates heketi volume in batches.
+
+        Args:
+            heketi_client_node (str): Node on which cmd has to executed.
+            heketi_server_url (str): Heketi server url.
+            vol_count (int): Count of volumes to create.
+            size (str): Volume size.
+            batch_amount (int): Amount of volumes to be created in one batch.
+            timeout (int): Timeout for entire volume count to create.
+            interval (int): Time interval between each interation.
+            skip_cleanup (bool): skip cleanup or not.
+
+        Returns:
+            list: list of heketi volume id's.
+
+        Kwargs:
+            The keys, values in kwargs are:
+                - name : (str)|None
+                - clusters : (str)|None
+                - ha : (int)|None
+                - auth : (bool)
+                - json : (bool)
+                - secret : (str)|None
+                - user : (str)|None
+        """
+        kwargs.pop("json", None)
+        h_vols, async_obj = [], []
+
+        # Fetch heketi volume list before operation
+        h_vols_initial = heketi_volume_list(
+            self.heketi_client_node, self.heketi_server_url,
+            json=True).get('volumes')
+
+        # Add cleanup function to clean stale volumes created during test
+        self.addCleanup(self.cleanup_heketi_volumes, h_vols_initial)
```

Comment:
Instead of adding this general cleanup function, check if new blockvolume creation leads to create BHV and add cleanup to delete newly created BHV.

Reply:
Done